src/remotes.js

```diff
@@ -97,48 +95,29 @@ angular.module('ostree.remotes', [
         }
 
         function loadRemoteSettings(name) {
-            var file = getRemoteSettingsFile(name);
-            var section = getSectionName(name);
-            var d = $q.defer();
-            file.read()
-                .done(function (content) {
-                    var data = parseData(content);
-                    if (data[section])
-                        d.resolve(data[section]);
-                    else
-                        d.reject(_("No configuration data found"));
-                })
-                .fail(function (ex) {
-                    d.reject(ex);
-                })
-                .always(function () {
-                    file.close();
-                });
-
-            return d.promise;
+            const file = getRemoteSettingsFile(name);
+            const section = getSectionName(name);
+            return new Promise((resolve, reject) => {
+                file.read()
+                    .then(content => {
+                        const data = parseData(content);
+                        if (data[section])
+                            resolve(data[section]);
+                        else
+                            reject(_("No configuration data found"));
+                    })
+                    .fail(reject)
+                    .always(file.close);
+            });
         }
 
         function updateRemoteSettings(name, options) {
-            var file = getRemoteSettingsFile(name);
-            var section = getSectionName(name);
-            var d = $q.defer();
-
-            function mutate(content) {
-                return changeData(content, section, options);
-            }
-
-            file.modify(mutate)
-                .done(function () {
-                    d.resolve();
-                })
-                .fail(function (ex) {
-                    d.reject(ex);
-                })
-                .always(function () {
-                    file.close();
-                });
+            const file = getRemoteSettingsFile(name);
+            const section = getSectionName(name);
 
-            return d.promise;
+            const promise = file.modify(content => changeData(content, section, options));
```

Comment:
Did you miss the fail handler here?

Reply:
Ah actually it's trivial, not doing anything - sorry for noise.

Reply:
Right, the original code just passed it on and just re-wrapped the rejection into a new defer.reject(). I simplified the code to simply pass on the original promise instead, and just kept the .always(file.close) cleanup.

Reply:
Please do bring up the "noise" -- that's the point of reviews, to double-check :-) Thanks!